### PR TITLE
!MERGE - Feature/purge api responses

### PR DESCRIPTION
**Pull request by jonsherrard**
#### What have you done
Rewriting the cache manager so that a central cache object is queryable for various statuses.

#### Why have you done it
Intergation with WordPress requires that we respond with sensible status messages, purge complete, purge failed, nothing to purge etc. The current system of broadcasting events from various caches and endpoints is likely to get messy.

#### Testing carried out to prevent breaking changes
Existing tests cover a bit, but need further work.

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ import handlePurge from './handle-purge'
 import handleRedirects from './handle-redirects'
 import CacheManager from '../utilities/cache-manager'
 
+const cacheManager = new CacheManager
```
Comment (thomasdigby) [inline src/server/index.js]: Did you mean for this to be `new CacheManager()`?

Comment (jonsherrard) [inline src/server/index.js]: Yes. I also don't really like the implementation of new.

I just want to import `cacheManager from 'cacheManager'`... which might work now actually.

Comment (thomasdigby) [inline src/server/index.js]: Can you run me through it before 10? I'd be interested to see it up and running

**Comment by thomasdigby** [comment]
What's the status of this PR? Do we still have bits to do?

**Pull request merged**